### PR TITLE
Cloud: retrieve up-to-date visibility information before saving existing score

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #13462
Resolves: #13571

Builds on: #13558

### CODE REVIEW

File: src/project/internal/projectactionscontroller.cpp [hunk 1]
```diff
@@ -501,11 +501,32 @@ bool ProjectActionsController::saveProjectToCloud(const CloudProjectInfo& info,
...
-    if (isCloudAvailable && project->isNewlyCreated()) {
-        Ret ret = openAudioGenerationSettings();
-        if (!ret) {
+    bool generateAudio = false;
+
+    bool isPublic = info.visibility == CloudProjectVisibility::Public;
+
+    if (saveMode == SaveMode::Save && isCloudAvailable) {
+        // Get up-to-date visibility information
+        auto scoreInfo = cloudProjectsService()->downloadScoreInfo(info.sourceUrl);
```
Comment (RomanPudashkin) [inline src/project/internal/projectactionscontroller.cpp:510]: please don't use auto here, and don't ignore possible errors, print them to the log

**Review by RomanPudashkin — APPROVED** [review]

**Comment by shoogle** [comment]
MuseScore.com has the ability to make scores Unlisted as well as Public or Private. Will this preserve the Unlisted status?

@bkunda doesn't want to offer Unlisted as an option in the UI for Save/Publish to cloud, but it should be preserved if the user does Ctrl+S on a score that was separately made Unlisted online.

**Comment by bkunda** [comment]
> MuseScore.com has the ability to make scores Unlisted as well as Public or Private. Will this preserve the Unlisted status?
> 
> @bkunda doesn't want to offer Unlisted as an option in the UI for Save/Publish to cloud, but it should be preserved if the user does Ctrl+S on a score that was separately made Unlisted online.

@cbjeukendrup Please let me know when you've made this amendment, and I'll then give this PR a test :-)

**Comment by shoogle** [comment]
@bkunda, how about we test and merge this as it is and deal with Unlisted scores after the beta?

**Comment by cbjeukendrup** [comment]
I rebased it so that it is mergeable at all.

I hope to fix "unlisted" tonight.

**Comment by bkunda** [comment]
> I rebased it so that it is mergeable at all.
> 
> I hope to fix "unlisted" tonight.

Thanks @cbjeukendrup! We can look at this when you're ready then :-)

**Comment by cbjeukendrup** [comment]
@bkunda @shoogle @RomanPudashkin I implemented handling for unlisted scores. Please recheck :)

**Review by RomanPudashkin — APPROVED** [review]

**Comment by cbjeukendrup** [comment]
@bkunda Next round: the issue you had mentioned to me should be fixed now!

**Comment by cbjeukendrup** [comment]
@bkunda Now that the API has been updated, and the code of this PR accordingly, the "unlisted" setting should be preserved correctly!

**Comment by bkunda** [comment]
Hey @cbjeukendrup just tested on macOS and everything seems to be working as expected! 🙂👍🏻🙏🏻

Only issue I had was when trying to test in Windows: I couldn't actually log in to musescore.com (I got as far as the new web page saying "you're good to go back to the desktop app", but the desktop app never seemed to recognise this), hence I haven't been able to test in Windows. 

Perhaps @shoogle can just sanity check this in Windows before we give it the green light? I haven't yet heard of any issues with Windows users being unable to log into their accounts though, so maybe it's just a quirk of this PR on my particular set up. 🤷‍♂️

**Comment by cbjeukendrup** [comment]
@bkunda It might be interesting to see the logs; maybe they contain a clue about where it goes wrong. Could you upload them?

**Pull request merged**